pinc/smoothread.inc

```diff
@@ -194,14 +194,24 @@ function handle_smooth_reading_change($project, $postcomments, $days, $extend)
     $projectid = $project->projectid;
 
     if ($days) { // will be 0 if parameter not supplied when only replacing text
-        $seconds = $days * 60 * 60 * 24;
         if ($extend) {
             // extend deadline if not yet passed
-            $deadline = $project->smoothread_deadline + $seconds;
+            $start_time = $project->smoothread_deadline;
         } else {
             // if starting sr with deadline=0, or if sr ended
-            $deadline = time() + $seconds;
+            $start_time = time();
         }
+
+        // Extend the deadline to the next whole hour after $days
+        $datetime = new DateTime("@$start_time");
+        // Add our days
+        $datetime->modify("+{$days} day");
+        // round up to the next hour
+        $datetime = $datetime->modify("next hour");
```

Comment:
Not very important but this doesn't actually do what I'd intended. If it is already a whole hour it increases to the next. To avoid this we could add 
```
$datetime = $datetime->modify("previous second");
```
before this line.

Reply:
I think it's fine as-is -- I don't see that an extra hour when we're setting things in days is going to be a big deal.

Reply:
I saw that when looking at one of the test projects. I don't see it as a big deal.